Dockerfile.dev

```diff
@@ -6,14 +6,14 @@ RUN npm install -g pnpm@7.5.2
 
 # Setup proxy to API used in saleor-platform
 RUN apt-get update && apt-get install -y nginx
-COPY ./nginx/dev.conf /etc/nginx/conf.d/default.conf
+COPY apps/storefront/nginx/dev.conf /etc/nginx/conf.d/default.conf
 
-COPY package.json ./
-COPY pnpm-lock.yaml ./
-RUN pnpm install
 COPY . .
-ARG API_URI
-ENV API_URI ${API_URI:-http://localhost:8000/graphql/}
+RUN pnpm install
+
+
+ARG SALEOR_API_URL
+ENV SALEOR_API_URL ${SALEOR_API_URL:-http://localhost:8000/graphql/}
 
 EXPOSE 3000
-CMD pnpm dev -- --hostname 0.0.0.0
+CMD pnpm turbo run dev --parallel --cache-dir=.turbo
```

Comment:
I'm wondering, why do we use `dev` command instead of `build` + `start`?

Reply:
@magul this file is named `Dockerfile.dev` - dev which means is's for development? 🤔 Shouldn't we use something else for local development and Docker for production deployments?

In this configuration if you change something in the project's code you'd still have to re-build the docker image on every change. If that's our use case then maybe we should use [bind mounts](https://docs.docker.com/storage/bind-mounts/) instead of `COPY`

Reply:
Maybe we should have 2 images, one development with volums, second production with COPY ?

Reply:
yup, for production `COPY` makes more sense

Reply:
@mmiszy:
> I'm wondering, why do we use `dev` command instead of `build` + `start`

I believe that will lower an entry level for exploring storefront, when fetched for the fist time inside `saleor-platform`.
My understanding of a purpose of `saleor-platform` is that you can just fetch it, build it, start it and start playing immediately with a source code (which will be shared between docker container and host machine using volumes).
Having `dev` instead `build` and `start` allow it to be hot-reloaded (as I correctly understand, and assuming that socket responsible for that will be also available from outside of a container).

@witoszekdev @lkostrowski  answering all your comments:
 * I need to `COPY` these assets during image build phase, as I need to be able to install npm dependencies there; we can have a conversation whether we want that to be done in entrypoint and making startup of these apps slower there (just be aware that I'm not changing this behaviour - it worked that way in pre-mono epoch also)
 * bind volumes were/are used in that way in pre-mono setup and I do intend to use them the same way (see: https://github.com/saleor/saleor-platform/blob/main/docker-compose.yml#L46, it will require changes, but I want to deliver that after figuring out how to encapsulate new, mono- storefront in docker)
 * beyond `saleor-platform` I dont' recall other development-centered usage for that `Dockerfile`, we could have a conversation about production-grade docker image, but I  don't know if we accutaly encorage people to use it as it is in their projects and want to keep that out of scope of this pull request (as that's yet another major conversation, how we want to support our users in that regard). One thing - is it used in demo e.g.? @witoszekdev maybe somebody from the cloud team know, what do we use there and how it's built?

Reply:
@magul Storefront repo is deployed on Vercel, it's not using Dockerfile in production. The `demo.saleor.io/graphql` instance is managed by cloud team, I don't know whay they use for that 😅 - it's special because it masks PII and the dashboard is read only.

Reply:
>  Storefront repo is deployed on Vercel
So maybe we don't need to provide production-grade docker image for that?

> demo.saleor.io/graphql
there's GraphQL playground/endpoint, but I asking about storefront that consumes that at https://demo.saleor.io/default-channel/en-US ? Is that this `react-storefront`? If cloud team support it and there's SSR involved, so they should have a way to deploy that (as we use Kubernetes, I'm assuming there's a `Dockerfile` somewhere). Also - are they aware that monorepo change and deployment pipeline reflects that already?

Reply:
[Here](https://github.com/saleor/saleor-platform/discussions/166) is discussion thread we can have regarding deployment strategy. At the moment let's fix local dev setup

Reply:
> Here is discussion thread we can have regarding deployment strategy. At the moment let's fix local dev setup

That's even more complex topic, as (as far as I understand) storefront is something unique for a store, so don't know if it should be considered as fixed part of Saloer suite of things that you want to deploy.
But yes, let's park that conversation for now.

Reply:
I've just read conversation about issue in CLI regarding deploying storefront, so I'm withdrawing that claim:
> That's even more complex topic, as (as far as I understand) storefront is something unique for a store, so don't know if it should be considered as fixed part of Saloer suite of things that you want to deploy.

Silly me!